File: Rishu-PB002/wiki/wiki.php
<?php
error_reporting(E_ALL ^ E_WARNING); 
if (count($argv) == 1) {
	echo "Enter a query : ";
	$topic = fgets(STDIN);
	echo "Enter a file name : ";
	$fileName = fgets(STDIN);
	queryfunction($topic, $fileName);
} else { 
	$fileName=$argv[1];
	$topic=$argv[2];
	for ($i=3;$i<count($argv);$i++){
		$topic=$topic."%20".$argv[$i];
	}
	queryfunction($topic, $fileName);
}


function queryfunction($topic, $fileName){
	$topic= str_replace(' ', '%20', $topic); 
	$url = "https://en.wikipedia.org/w/api.php?action=opensearch&limit=1&format=json&search=".$topic;
	
	try {
		$json = file_get_contents($url);
		$json_data = json_decode($json, true);
		foreach ($json_data as $k=>$v){
		}
		$final_url= $v[0]; 
		$date= date('Y-m-d H:i:s');
		file_put_contents($fileName, print_r($final_url." ".$date."\n", true), FILE_APPEND);
		echo($final_url);
	} 
	catch (Exception $e) {
    	echo 'Error: ',  $e->getMessage(), "\n";
	}
	
}

?>

File: Rishu-PB002/wiki/wiki.py
import requests
import json
import logging
from requests.exceptions import ConnectionError
import sys
import datetime


def queryfunction(query, file):

    api_url = "https://en.wikipedia.org/w/api.php?action=opensearch&limit=1&format=json&search="+query

    try:
        myResponse = requests.get(api_url)
        if(myResponse.ok):
            final_key=""
            jData = json.loads(myResponse.content)
            for key in jData:
                final_key=json.dumps(key)
        url = final_key[2:-2]
        logging.basicConfig(filename=file,level=logging.DEBUG)
        logging.info(str(datetime.datetime.now())+ "   " +url)
        print url

    except ConnectionError as e:
        r = "Error Connecting"
        print r

    

#if command line arguments are present 2nd is query with no space 1st is filename    
try:
    
    filename=sys.argv[1]
    query=sys.argv[2]
    for i in range(3,len(sys.argv)):
        query=query+" "+sys.argv[i];
    queryfunction(query,filename)
    
except:
    query = raw_input("Please enter your search query for wikipedia : ")
    file= raw_input("Please enter the filename : ")
    file=file+".log"
    queryfunction(query,file)
    
